cmplxs/cmplxs.go

```diff
@@ -0,0 +1,669 @@
+func CumSum(dst, s []complex128) []complex128 {
+	if len(dst) != len(s) {
+		panic("cmplxs: length of destination does not match length of the source")
+	}
+	if len(dst) == 0 {
+		return dst
+	}
+	return c128.CumSum(dst, s)
+}
+
+// Distance computes the L-norm of s - t. See Norm for special cases.
+// A panic will occur if the lengths of s and t do not match.
+func Distance(s, t []complex128, L float64) float64 {
+	if len(s) != len(t) {
+		panic("cmplxs: slice lengths do not match")
+	}
+	if len(s) == 0 {
+		return 0
+	}
+	if L == 2 {
+		return c128.L2DistanceUnitary(s, t)
+	}
+	var norm float64
+	if L == 1 {
+		for i, v := range s {
+			norm += cmplx.Abs(t[i] - v)
+		}
+		return norm
+	}
+	if math.IsInf(L, 1) {
+		for i, v := range s {
+			absDiff := cmplx.Abs(t[i] - v)
+			if absDiff > norm {
+				norm = absDiff
+			}
+		}
+		return norm
+	}
+	for i, v := range s {
+		norm += math.Pow(cmplx.Abs(t[i]-v), L)
+	}
+	return math.Pow(norm, 1/L)
+}
+
+// Div performs element-wise division dst / s
+// and stores the value in dst. It panics if the
+// lengths of s and t are not equal.
+func Div(dst, s []complex128) {
+	if len(dst) != len(s) {
+		panic("cmplxs: slice lengths do not match")
+	}
+	c128.Div(dst, s)
+}
+
+// DivTo performs element-wise division s / t
+// and stores the value in dst. It panics if the
+// lengths of s, t, and dst are not equal.
+func DivTo(dst, s, t []complex128) []complex128 {
+	if len(s) != len(t) || len(dst) != len(t) {
+		panic("cmplxs: slice lengths do not match")
+	}
+	return c128.DivTo(dst, s, t)
+}
+
+// Dot computes the dot product of s1 and s2, i.e.
+// sum_{i = 1}^N s1[i]*s2[i].
+// A panic will occur if lengths of arguments do not match.
+func Dot(s1, s2 []complex128) complex128 {
+	if len(s1) != len(s2) {
+		panic("cmplxs: lengths of the slices do not match")
+	}
+	return c128.DotUnitary(s1, s2)
+}
+
+// Equal returns true if the slices have equal lengths and
+// all elements are numerically identical.
+func Equal(s1, s2 []complex128) bool {
+	if len(s1) != len(s2) {
+		return false
+	}
+	for i, val := range s1 {
+		if s2[i] != val {
+			return false
+		}
+	}
+	return true
+}
+
+// EqualApprox returns true if the slices have equal lengths and
+// all element pairs have an absolute tolerance less than tol or a
+// relative tolerance less than tol.
+func EqualApprox(s1, s2 []complex128, tol float64) bool {
+	if len(s1) != len(s2) {
+		return false
+	}
+	for i, a := range s1 {
+		if !EqualWithinAbsOrRel(a, s2[i], tol, tol) {
+			return false
+		}
+	}
+	return true
+}
+
+// EqualFunc returns true if the slices have the same lengths
+// and the function returns true for all element pairs.
+func EqualFunc(s1, s2 []complex128, f func(complex128, complex128) bool) bool {
+	if len(s1) != len(s2) {
+		return false
+	}
+	for i, val := range s1 {
+		if !f(val, s2[i]) {
+			return false
+		}
+	}
+	return true
+}
+
+// EqualWithinAbs returns true if parts of a and b have an absolute
```

Comment:
`parts of a and b` doesn't match the use of `cmplx.Abs(a-b)`. I prefer to change the comment here rather than the implementation.

Reply:
Done